ocaml/xapi/pool_recommendations.ml

```diff
@@ -0,0 +1,44 @@
+(*
+ * Copyright (c) Cloud Software Group, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module Unixext = Xapi_stdext_unix.Unixext
+module Config_file = Xcp_service.Config_file
+
+module D = Debug.Make (struct let name = "pool_recommendations" end)
+
+open D
+module StringMap = Map.Make (String)
+
+let process_line data map =
+  match Config_file.parse_line data with
+  | Some (key, v) ->
+      debug "Parsing data, key: %s, value: %s" key v ;
+      StringMap.add key v map
+  | None ->
+      map
+
+let parse filename map =
+  debug "Parsing pool recommendations file: %s" filename ;
+  Unixext.file_lines_fold (fun m line -> process_line line m) !map filename
+
+let load rec_dir =
+  let rec_map = ref StringMap.empty in
+  (try Sys.readdir rec_dir with _ -> [||])
+  |> Array.to_list
+  |> List.stable_sort compare
+  |> List.iter (fun file ->
+         let path = Filename.concat rec_dir file in
+         rec_map := parse path rec_map
+     ) ;
+  StringMap.bindings !rec_map
```

Comment:
We don't need a ref type here. Just `List.fold_left` with `StringMap.empty` as starting value.